src/main.lisp

```diff
@@ -0,0 +1,64 @@
+(in-package :cl-repl)
+
+(defconstant +version+ '0.4.0)
+
+(defvar *logo*
+"  ___  __          ____  ____  ____  __
+ / __)(  )    ___ (  _ \\(  __)(  _ \\(  )
+( (__ / (_/\\ (___) )   / ) _)  ) __// (_/\\
+ \\___)\\____/      (__\\_)(____)(__)  \\____/
+")
+
+(defvar *copy* "(C) 2017-2018 TANI Kojiro <kojiro0531@gmail.com>")
+
+(defvar *versions*
+  #+ros.script
+  (format nil "ver. ~a on ~?~a ~a"
+    +version+
+    #+ros.script
+    "Roswell ~a, "
+    #-ros.script
+    ""
+    #+ros.script
+    `(,(ros::version))
+    #-ros.script
+    nil
+    (lisp-implementation-type)
+    (lisp-implementation-version)))
+
+(defmacro when-option ((options opt) &body body)
+  `(let ((it (getf ,options ,opt)))
+     (when it
+       ,@body)))
+
+(opts:define-opts
+  (:name :help
+         :description "Print this help and exit."
+         :short #\h
+         :long "help")
+  (:name :version
+         :description "Show the version info and exit."
+         :short #\v
+         :long "version"))
+
+(defun main (argv &key (show-logo t))
+  (multiple-value-bind (options free-args)
```

Comment:
nice ! I like this way of parsing arguments. If by chance my code helped, I'm glad it did :)

Reply:
Right, thanks a lot!
However, I could not build an executable with asdf ...
To be accurate, the build succeeded, but the executable did not work, caused SB-SYS: MEMORY-FAULT-ERROR.
I'm not sure where this error comes.
Do you have any ideas?

Reply:
No, I didn't encounter this error. I had once a "heap exhausted error" and it was a recursion error in a (new-ish) third-party library.

If you put your attempt in a new branch I'm ok to try, and at least say if I reproduce the pb.

Reply:
OK, please.
I'll create the branch and put comment on #19 later.